### PR TITLE
fix: Erroneous number of search results when using advanced people search - EXO-63148 - Meeds-io/meeds#818

### PR DESCRIPTION
**Pull request by akhanfir**
Prior to this change, when go to People page and make an advanced search on a field and confirm, The number of results displayed on the top of the page is less than the real number of cards displayed by 1 card. This problem occurs when the current user is not included in this list. To fix that, it is necessary to calculate the correct size of the list of people in backend by adding the current user in excludedIdentityList of filter.
After this changes, the number of results displayed is the size of the list of all the people who are involved in this search except the current user who should not be applied in this search.

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 1]
```diff
@@ -454,6 +454,11 @@ public Response getUsersOrConnectionsByAdvancedFilter(@Context UriInfo uriInfo,
           filter.setUserType(userType);
         }
       }
+      if (settings != null) {
+        List<Identity> excludedIdentityList = new ArrayList();
+        excludedIdentityList.add(target);
+        filter.setExcludedIdentityList(excludedIdentityList);
```
Comment (ahamdi) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Could this be improved like this, WDYT?
```suggestion
        filter.setExcludedIdentityList(Collections.singletonList(target));
```

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you rebase the PR and fix conflict

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 2]
```diff
@@ -315,9 +315,13 @@ public Response getUsers(@Context UriInfo uriInfo,
...
       }
     } else {
+      Identity target = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, userId);
       ProfileFilter filter = new ProfileFilter();
       filter.setName(q == null || q.isEmpty() ? "" : q);
       filter.setEnabled(!isDisabled);
+      if (target != null && filter.getName().equals(q)) {
+        filter.setExcludedIdentityList(Collections.singletonList(target));
+      }
       if (!isDisabled) {
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: @ahamdi The getUsers function (and the restService in general) provide the list of users which match all parameters. There is no information about excluding the current user.
I'm afraid that excluding the current user in the rest service can lead to problems in other parts of the platform, where the current user is needed.

If this task is too complex to fix, we can postpone it to 6.4.1.

I think we should take reflexion time here

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java (hunk 2 above)
Comment (ahamdi) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: @rdenarie ok I see the point
we can add a new optional parameter to filter or not the connected user WDYT ?

Comment (rdenarie) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Yes I think it is the better solution.
But is it possible for 6.4.0 ? or is it safer to do it properly in 6.4.1 ?

Comment (ahamdi) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: @rdenarie Yes very possible for 6.4.0

Comment (rdenarie) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: ok, so go with a new parameter in the function. 

I keep the possibility to push back in 6.4.1 if not fully finished at the end of the week

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: component/service/src/main/java/org/exoplatform/social/rest/api/UserRestResources.java [hunk 3]
```diff
@@ -49,7 +49,8 @@ public abstract Response getUsers(@Context UriInfo uriInfo,
                                     @QueryParam("offset") int offset,
                                     @QueryParam("limit") int limit,
                                     @QueryParam("returnSize") boolean returnSize,
-                                    @QueryParam("expand") String expand) throws Exception;
+                                    @QueryParam("expand") String expand,
+                                    @QueryParam("excludedIdentity") boolean excludedIdentity) throws Exception;
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/api/UserRestResources.java]:
```suggestion
                                    @QueryParam("excludeCurrentUser") boolean excludeCurrentUser) throws Exception;
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — DISMISSED** [review]

**Comment by rdenarie** [comment]
@akhanfir there is a build problem. Can you check ? 
I think there is a comment to update in PeopleRestService

**Review by rdenarie — APPROVED** [review]

**Pull request merged**